src/TemplateSkinManager.php

```diff
@@ -0,0 +1,122 @@
+<?php
+/**
+ * Handle form template view skin.
+ *
+ * @package Give
+ */
+
+namespace Give;
+
+/**
+ * Class TemplateSkinManager
+ *
+ * @since 2.7.0
+ * @package Give
+ */
+class TemplateSkinManager {
+	/**
+	 * Document page title.
+	 *
+	 * This will be use to setup title tag.
+	 *
+	 * @since 2.7.0
+	 * @var string
+	 */
+	protected $title;
+
+	/**
+	 * Document page body.
+	 *
+	 * This will be use to setup content of body tag.
+	 *
+	 * @since 2.7.0
+	 * @var string
+	 */
+	protected $body;
+
+	/**
+	 * Body classes.
+	 *
+	 * This will be use to setup body tag classes.
+	 *
+	 * @since 2.7.0
+	 * @var array
+	 */
+	protected $bodyClasses = [ 'give-form-templates' ];
+
+	/**
+	 * Set document page title.
+	 *
+	 * @param string $title
+	 *
+	 * @return TemplateSkinManager $this
+	 */
+	public function setTitle( $title ) {
+		$this->title = $title;
+
+		return $this;
+	}
+
+	/**
+	 * Set document page title.
+	 *
+	 * @param string $body
+	 *
+	 * @return TemplateSkinManager $this
+	 */
+	public function setBody( $body ) {
+		$this->body = $body;
+
+		return $this;
+	}
+
+	/**
+	 * Set document page title.
+	 *
+	 * @param array $classes
+	 *
+	 * @return TemplateSkinManager $this
+	 */
+	public function setBodyClasses( $classes ) {
+		$this->bodyClasses = array_merge( $this->bodyClasses, (array) $classes );
+
+		return $this;
+	}
+
+
+	/**
+	 * Render view.
+	 *
+	 * Note: if you want to overwrite this function then do not forget to add action hook in footer and header.
+	 * We use these hooks to manipulated donation form related actions.
+	 *
+	 * @since 2.7.0
+	 */
+	public function render() {
```

Comment:
Let's just write to a string and return that. Ideally, we have a single place that compiles the views and writes to the output buffer. Think of the output buffer as a global. If we just write to it here and there we'll end up with rather unpredictable results. Make sense?

Reply:
yes, I will return result from this function.

Reply:
resolved